src/Razor/test/Microsoft.AspNetCore.Razor.Test.Common/ThrowingTraceListener.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Razor.Test.Common;
+
+public sealed class ThrowingTraceListener : TraceListener
+{
+    public static void ReplaceDefaultListener()
+    {
+        // Remove the default trace listener so that Debug.Assert and Debug.Fail don't diplay
+        // assert dialog during test runs.
+
+        var sawThrowingTraceListener = false;
+        var listeners = Trace.Listeners;
+        for (var i = listeners.Count - 1; i >= 0; i--)
+        {
+            switch (listeners[i])
+            {
+                case DefaultTraceListener:
+                    listeners.RemoveAt(i);
+                    break;
+                case ThrowingTraceListener:
+                    sawThrowingTraceListener = true;
+                    break;
+            }
+        }
+
+        if (!sawThrowingTraceListener)
+        {
+            // Add an instance of the ThrowingTraceListener so that Debug.Assert and Debug.Fail
+            // throw exceptions during test runs.
+            listeners.Add(new ThrowingTraceListener());
+        }
+    }
+
+    public override void Fail(string? message, string? detailMessage)
```

Comment:
From my understanding, a single test will fail with this exception now instead of hanging the machine right?

Reply:
Actually, I'll do this a bit differently. The `DefaultTraceListener` has some useful behavior too, since it controls `Debug.Write(...)` as well. I'm thinking that I'll leave it in place but disable the assertion UI.